app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,23 @@
 class PostsController < ApplicationController
   def index; end
 
-  def show; end
+  def show
+    @post = Post.where(id: params[:id])[0]
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    author = User.find(params['post']['author_id'].to_i)
+    title = params['post']['title']
+    text = params['post']['text']
```

Comment:
- [ ] **[OPTIONAL]** I suggest you make use of the strong parameters in all your controllers instead of getting the params as you have here. You can check this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) to learn about strong parameters.